File: Dave Min/2023-05-10-Linked_List_Cycle.cs
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public boolean hasCycle(ListNode head) {
        if(head ==null || head.next == null) return false;
        ListNode hare = head.next;
        ListNode tortoise = head;
        while(hare!=null && hare.next!=null ){
            if(hare == tortoise) return true;
            hare = hare.next.next;
            tortoise = tortoise.next;
        }
        return false;
    }
}

//TC: O(N)
//SC: O(1)

File: Ellen Jung/2023-05-10-linked-list-cycle.py
# Definition for singly-linked list.
class ListNode(object):
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution(object):
    def hasCycle(self, head):
        """
        :type head: ListNode
        :rtype: bool
        """
        list = []
        while head.next:
            if head.next.val in list:
                return True
            else:
                list.append(head.next.val)
                head = head.next
        return False